### PR TITLE
Test private protected scopes

### PR DESCRIPTION
**Pull request by Jurigag**
Hello!

* Type: code quality
* Link to issue:

In raising this pull request, I confirm the following:

- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [ ] I updated the CHANGELOG

Small description of change:

Thanks

### CODE REVIEW

File: unit-tests/Extension/Oo/Scopes/PrivateScopeTest.php [hunk 1]
```diff
@@ -13,12 +13,175 @@
...
+    }
+
+    /**
+     * @expectedException \Error
+     */
+    public function testShouldNotSetPrivatePropertyExtendedMagicObjPhp()
+    {
+        $obj = new TestScopePhpMagicExtending();
+        $tester = new PropertyTester();
+        $tester->setPropertyObj($obj, 'privateProperty2', 'test');
```
Comment (AlexNDRmac) [inline unit-tests/Extension/Oo/Scopes/PrivateScopeTest.php:154]: Could you please use DataProviders for similar tests

File: unit-tests/Extension/Oo/Scopes/PrivateScopeTest.php [hunk 2]
```diff
@@ -13,12 +13,175 @@
...
+        $this->assertFalse(true, 'This should not be called');
+    }
+
+    /**
+     * @expectedException \Error
```
Comment (AlexNDRmac) [inline unit-tests/Extension/Oo/Scopes/PrivateScopeTest.php:159]: For exceptions test - better is to use [built-in phpunit assertion](https://phpunit.readthedocs.io/en/latest/writing-tests-for-phpunit.html#testing-exceptions) `$this->expectedException()`

File: unit-tests/Extension/Oo/Scopes/PrivateScopeTest.php (hunk 1 above)
Comment (Jurigag) [reply unit-tests/Extension/Oo/Scopes/PrivateScopeTest.php:154]: Surely, for now just wanted to check all use cases, will refactor this thing later.

**Comment by sergeyklay** [comment]
Closed in favor of https://github.com/phalcon/zephir/pull/2078

**Pull request closed without merging**